src/x509.c

```diff
@@ -385,6 +385,24 @@ static LUA_FUNCTION(openssl_x509_check)
   }
 }
 
+static LUA_FUNCTION(openssl_x509_check_host)
+{
+#if OPENSSL_VERSION_NUMBER > 0x10002000L
```

Comment:
please move #if OPENSSL_VERSION_NUMBER > 0x10002000L out of openssl_x509_check_host,